### PR TITLE
Moved imports into functions that use them

### LINKED ISSUE #8327: GBQ: Handle deprecation of oauth2client.tools.run()
Right now, when you use GbqConnector, you get the following warning:

>WARNING:root:This function, oauth2client.tools.run(), and the use of the gflags library are deprecated and will be removed in a future version of the library.

I'm not sure how quickly this deprecation will happen or if there are alternatives to keep things the way that they are; but at least worthwhile to have an issue here to address it (in the future).

As an aside, I've been glancing at the source for the `bq` client that Google provides as part of Google Cloud SDK and I'm wondering how public the methods are (i.e., if we could adapt some of the code or functions that `bq.py` uses to authenticate and be able to get credentials set up from `gcloud auth login` rather than requiring a separate oauth flow). Not sure though.

Again, cc @jacobschaer , I'd be interested in your opinion on whether this is worth investigating at all. I've been finding `read_gbq` and `to_gbq` quite useful - thanks for putting it together!

### LINKED ISSUE #9713: On demand imports failing in pandas.io.gbq
After upgrading to 0.16.0 BigQuery functionality is not working. Upon invoking pd.read_gbq(), the _test_imports() function is running, which in turn runs _importers(). Everything is all well and good up to here, but per pdb, once _importers() returns, the imported modules are not accessible and read_gbq() throws ```NameError: global name 'OAuth2WebServerFlow' is not defined```.

### PR DESCRIPTION
**Pull request by cgrin**
closes #9713
closes #8327

Changes to be committed:
modified: pandas/io/gbq.py

Moved google api imports into functions that call them, removed unused imports. This fixes #9713

### CODE REVIEW

**Comment by cgrin** [comment]
cc @jreback @jacobschaer

I've tested this change and successfully pulled data from and pushed to GBQ with it.

File: pandas/io/gbq.py [hunk 1]
```diff
@@ -13,75 +13,22 @@
...
-            except ImportError:
-                _GOOGLE_API_CLIENT_INSTALLED = False
-
-
-            try:
-                import gflags as flags
-                _GOOGLE_FLAGS_INSTALLED = True
-
-                _GOOGLE_FLAGS_VERSION = pkg_resources.get_distribution('python-gflags').version
+if compat.PY3:
```
Comment (jreback) [inline pandas/io/gbq.py]: this needs to be moved into a function as this will raise on import of pandas when under py3.

**Comment by cgrin** [comment]
@jreback good point. I've moved the Python3 check into the function that checks for compatibility with the google API

**Comment by jreback** [comment]
- can you rebase
- pls add a release note to 0.16.1

**Comment by jreback** [comment]
cc @jacobschaer

can you give a test out pls.

**Comment by cgrin** [comment]
Rebase done and release note added.

**Comment by jreback** [comment]
@cgrin looks like you merged in something else.

see instruction on git-fu [here](http://pandas-docs.github.io/pandas-docs-travis/contributing.html)

**Comment by cgrin** [comment]
Alright, that should take care of that. Not sure what happened yesterday. This closes #9713 and #8327

**Comment by jreback** [comment]
@cgrin the ``_test_imports`` is still called somewhere (and is failing)

**Comment by cgrin** [comment]
@jreback Looks like it's called in the test suite. Since the _test_imports function itself gone due to the imports happening on demand, should we remove this test? It's the test_requirements function (lines 32-36 of pandas/io/tests/test_gbq.py) raising the error, which is in turn called by each class in the file.

**Comment by cgrin** [comment]
We could also bring that function into test_gbq.py, removing the imports that are no longer necessary.

**Comment by jreback** [comment]
you *can* change the tests. The idea is to have a series of tests:

- that will allow importing of gbq and not call anything (e.g. to test that if the deps are not installed simply importing will not fail)
- use ``to/read_gbq`` with deps installed (which IIRC the ci does), but since we don't have credentials, they skip gracefully (e.g. the expected expections happen)
- 'real' testing, currently done offline only (as credentials are necessary) that actually tests functionaility

**Comment by cgrin** [comment]
Thanks. I opted to move this function into the test suite and remove the no longer needed imports. (Also rebased again as there has been some movement on the upstream repo). Hopefully this gets us where we need to be!

**Comment by cgrin** [comment]
Looks like I missed a couple imports there. Apologies, I'm having issues getting the tests to run at all locally. 686ee7e should fix the compatibility tests.

**Comment by jreback** [comment]
@cgrin ok, looks good. if you could squash to a single commit would be gr8.

anything beyond the release note for #8327. e.g. would a user seeing this need to do anything? 
this is prob not back-compat, but 0.16.0 is broken anyhow.....

**Comment by cgrin** [comment]
Rebased one more time and squashed everything.

Regarding #8327, the ```run``` function has been [deprecated since at least June 2013](https://groups.google.com/forum/#!topic/google-api-python-client/s4r7NRmKE3w), so it's very unlikely that anyone would need to take any action. ```google-api-python-client``` already has a version requirement of 1.2.0, which was released in July 2013 and includes the ```run_flow``` function, so this is more of a cleanup to get rid of the warning on auth than anything else.

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/2cf4132c6c30d0a26dd4fa316aaa7579a4fcc1d3

thanks!

**Pull request closed without merging**